### PR TITLE
LOOP-3130 Handling of alert acknowledgment failures

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-3130

### CODE REVIEW

File: DashKit/DashPumpManagerError.swift [hunk 1]
```diff
@@ -32,6 +33,17 @@ public enum DashPumpManagerError: Error, LocalizedError {
             return error.errorDescription
         case .busy:
             return LocalizedString("Pod Busy.", comment: "Description of DashPumpManagerError error when pump manager is busy.")
+        case .acknowledgingAlertFailed:
+            return LocalizedString("Tidepool Loop was unable to clear the alert on your Pod, therefore you may continue to hear an audible beep.", comment: "Description of DashPumpManagerError error when acknowledging alert failed.")
+        }
+    }
+    
+    public var recoverySuggestion: String? {
```
Comment (rickpasetto) [inline DashKit/DashPumpManagerError.swift:41]: Where is this referenced?  Is it in a later PR?

File: DashKit/DashPumpManager.swift [hunk 2]
```diff
@@ -1795,8 +1814,7 @@ extension DashPumpManager: PodSDKLoggingShimDelegate {
 
 // MARK: - AlertResponder implementation
 extension DashPumpManager {
-    public func acknowledgeAlert(alertIdentifier: Alert.AlertIdentifier) {
-        
+    public func acknowledgeAlert(alertIdentifier: Alert.AlertIdentifier, completion: @escaping (Error?) -> Void) {
```
Comment (rickpasetto) [inline DashKit/DashPumpManager.swift:1817]: Hm.  Is `DashPumpManager` actually an `AlertResponder`?  I don't see it declaring its conformance.  If it is, then this isn't going to build.  If it isn't, then, who calls this?

**Review by rickpasetto — COMMENTED** [review]
@ps2 Got a few questions first before proceeding...

File: DashKit/DashPumpManagerError.swift (hunk 1 above)
Comment (ps2) [reply DashKit/DashPumpManagerError.swift:41]: Oops, yeah, here's the Loop PR: https://github.com/tidepool-org/Loop/pull/435

File: DashKit/DashPumpManager.swift (hunk 2 above)
Comment (ps2) [reply DashKit/DashPumpManager.swift:1817]: DashPumpManager is a PumpManager, which conforms to DeviceManager, which conforms to AlertResponder

Comment (ps2) [reply DashKit/DashPumpManager.swift:1817]: Here's the LoopKit PR.  I thought I had created PRs for all of these, but somehow dropped that: https://github.com/tidepool-org/LoopKit/pull/387

**Review by rickpasetto — APPROVED** [review]
LGTM

**Pull request merged**